src/connectedk8s/azext_connectedk8s/_precheckutils.py

```diff
@@ -0,0 +1,220 @@
+        elif(dns_check == consts.Diagnostic_Check_Incomplete or outbound_connectivity_check == consts.Diagnostic_Check_Incomplete):
+            return consts.Diagnostic_Check_Incomplete, storage_space_available
+        else:
+            return consts.Diagnostic_Check_Failed, storage_space_available
+
+    # To handle any exception that may occur during the execution
+    except Exception as e:
+        logger.warning("An exception has occured while trying to execute cluster diagnostic checks container on the cluster. Exception: {}".format(str(e)) + "\n")
+        telemetry.set_exception(exception=e, fault_type=consts.Cluster_Diagnostic_Checks_Execution_Failed_Fault_Type, summary="Error occured while executing the cluster diagnostic checks container")
+
+    return consts.Diagnostic_Check_Incomplete, storage_space_available
+
+
+def executing_cluster_diagnostic_checks_job(corev1_api_instance, batchv1_api_instance, helm_client_location, kubectl_client_location, kube_config, kube_context, location, http_proxy, https_proxy, no_proxy, proxy_cert):
+    job_name = "cluster-diagnostic-checks-job"
+    # Setting the log output as Empty
+    cluster_diagnostic_checks_container_log = ""
+
+    cmd_helm_delete = [helm_client_location, "uninstall", "cluster-diagnostic-checks", "-n", "azure-arc-release"]
+    if kube_config:
+        cmd_helm_delete.extend(["--kubeconfig", kube_config])
+    if kube_context:
+        cmd_helm_delete.extend(["--context", kube_context])
+
+    # To handle the user keyboard Interrupt
+    try:
+        # Executing the cluster diagnostic checks job yaml
+        config.load_kube_config(kube_config, kube_context)
+        # Attempting deletion of cluster diagnostic checks resources to handle the scenario if any stale resources are present
+        response_kubectl_delete_helm = Popen(cmd_helm_delete, stdout=PIPE, stderr=PIPE)
+        output_kubectl_delete_helm, error_kubectl_delete_helm = response_kubectl_delete_helm.communicate()
+        # If any error occured while execution of delete command
+        if (response_kubectl_delete_helm != 0):
+            # Converting the string of multiple errors to list
+            error_msg_list = error_kubectl_delete_helm.decode("ascii").split("\n")
+            error_msg_list.pop(-1)
+            valid_exception_list = []
+            # Checking if any exception occured or not
+            exception_occured_counter = 0
+            for ind_errors in error_msg_list:
+                if('not found' in ind_errors or 'deleted' in ind_errors):
+                    pass
+                else:
+                    valid_exception_list.append(ind_errors)
+                    exception_occured_counter = 1
+            # If any exception occured we will print the exception and return
+            if exception_occured_counter == 1:
+                logger.warning("An error occured while installing the cluster diagnostic checks helm release in the cluster. Exception:")
```

Comment:
This warning and summary statement needs to be corrected. It should convey that cleanup of previous release failed and hence couldn't install the new diagnostic checks helm release. Please cleanup older release using "helm delete <release name> -n azuer-arc-release" and try again